libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/CycleStreetsRoutingTask.kt

```diff
@@ -7,9 +7,12 @@ import net.cyclestreets.view.R
 
 internal open class CycleStreetsRoutingTask(private val routeType: String,
                                             private val speed: Int,
-                                            context: Context) : RoutingTask<Waypoints>(R.string.route_finding_new, context) {
+                                            context: Context,
+                                            private val distance: Int? = null,
+                                            private val duration: Int? = null,
+                                            private val pois: String? = null) : RoutingTask<Waypoints>(R.string.route_finding_new, context) {
     override fun doInBackground(vararg waypoints: Waypoints): RouteData? {
         val wp = waypoints[0]
-        return fetchRoute(routeType, speed, wp)
+        return fetchRoute(routeType, -1, speed, wp, distance, duration, pois)
```

Comment:
I don't like that the interface change you've made forces us to put a magic number -1 here.
If it has to be so, please define a constant something like `NO_ITINTERARY = -1`, and use it both here and within the test in `doFetchRoute()`

Reply:
I'm pretty sure the -1 was already there in the previous code. (Probably in the FetchCycleStreetsRoutingTask.)  There were 2 fetchRoute functions and one just called the other, so it seemed tidier to combine them.

Happy to make a constant though.

Reply:
Indeed, we used an overload method for this. Outside the scope here makes it harder to see what's meant. And anyway, boy/girl scout rule of making things better as you go along, a descriptive constant would indeed be an improvement anyway 🙂